internal/sharedcomponent/sharedcomponent.go

```diff
@@ -85,20 +74,72 @@ func (c *Component[V]) Unwrap() V {
 
 // Start starts the underlying component if it never started before.
 func (c *Component[V]) Start(ctx context.Context, host component.Host) error {
+	if c.hostWrapper == nil {
+		c.hostWrapper = &hostWrapper{
+			host:           host,
+			sources:        make([]componentstatus.StatusReporter, 0),
+			previousEvents: make([]*componentstatus.StatusEvent, 0),
+		}
+	}
+
+	statusReporter, isStatusReporter := host.(componentstatus.StatusReporter)
+	if isStatusReporter {
+		c.hostWrapper.addSource(statusReporter)
+	}
```

Comment:
The new technique for sharedcomponent is to build the chain or reporting sources as Start is getting called.  Previously reported events are remembered so that if Start is called multiple times, each subsequent `StatusReporter` and get the previously emitted `StatusEvents`.